pkg/controller/elasticsearch/certificates/transport/csr.go

```diff
@@ -87,6 +87,8 @@ func buildGeneralNames(
 	generalNames := []certificates.GeneralName{
 		{OtherName: *commonNameOtherName},
 		{DNSName: commonName},
+		// add the transport service name for remote cluster connections initially connecting through the service
+		{DNSName: fmt.Sprintf("%s.%s.svc", esv1.TransportService(cluster.Name), cluster.Namespace)},
```

Comment:
We have `GetServiceDNSName()` in pkg/util/k8s which might be better to use here, and also includes the name without `.svc` (which also works)

Reply:
It will not work unless we also change the [seed hosts](https://github.com/elastic/cloud-on-k8s/blob/master/pkg/controller/elasticsearch/remotecluster/elasticsearch.go#L63)

Reply:
Makes sense, would a comment be helpful here to explain why we're not? It seems like a gotcha since the name resolves. Though to be fair, the fully qualified domain name with the default domain suffix (e.g. cluster.local) also resolves but would fail.